File: client/src/models/packets/packet-server-connection.js
import PacketLabels from '@/constants/packet-labels'

export default class PacketServerConnection {

  constructor(payload) {
    this.label = PacketLabels.SERVER_CONNECTION
    this.client = payload.client || null
    this.lobbyCode = payload.lobbyCode || null
  }

  receive(store) {
    store.setClient({ client: this.client })
    store.setLobbyCode({ lobbyCode: this.lobbyCode })
  }

}

File: server/models/packet_server_connection.go
package models

import (
	"encoding/json"
	"unlock/constants"
)

type PacketServerConnection struct {
	Packet
	Client    *Client `json:"client"`
	LobbyCode string  `json:"lobbyCode"`
}

func NewPacketServerConnection(client *Client, lobbyCode string) *PacketServerConnection {
	return &PacketServerConnection{
		Packet: Packet{
			Label: constants.PacketServerConnection,
		},
		Client:    client,
		LobbyCode: lobbyCode,
	}
}

func (packet *PacketServerConnection) Send(client *Client) (err error) {
	payload, err := json.Marshal(packet)
	if err != nil {
		return err
	}

	client.Channel <- payload

	return nil
}
